alicloud/resource_alicloud_slb_listener.go

```diff
@@ -369,6 +397,10 @@ func resourceAliyunSlbListenerCreate(d *schema.ResourceData, meta interface{}) e
 		return fmt.Errorf("WaitForListener %s got error: %#v", Running, err)
 	}
 
+	if isHttpRedirected {
+		return resourceAliyunSlbListenerRead(d, meta)
+	}
+
```

Comment:
Why need isHttpRedirected?

Reply:
with http redirected to https, the SetLoadBalancerHttpListenerAttribute is not supported, hence i have used that bool var to directly go for read. Please suggest if i have missed something in the approach.

Reply:
I think there is no need to set isHttpRedirected because of listener_forward and forward_port only can be effect when creating a new http listener. After that, there is no any change for them in the update method.

Reply:
@xiaozhu36 actually the way this resource is implemented is that after create, the update is also called and there are few defaults which then make update true and it goes for SetLoadBalancerHttpListenerAttribute api. Initially i had not used this bool but after getting the error, i used this bool to bypass update when creating for first time.

Reply:
So, if one listener is set HttpRedirected, it can not be updated whatever whether it is a new resource. Right?

Reply:
> So, if one listener is set HttpRedirected, it can not be updated whatever whether it is a new resource. Right?

Yes, that's correct. Update is not supported for http redirected listener

Reply:
ok, can you add a note for this limitation?

Reply:
@xiaozhu36 where should i put that note? i had kept a condition in update to give descriptive error to user but removed it as they were needless in a way.